### DISCUSSION TITLE
How to load connCompFile from Gamma products?

### DISCUSSION THREAD

**Discussion by qingyaol** [insarlab/MintPy#591] [Q&A] [answered]
I want to load the connected components files by customed,but I don't know the format of mask files.The supported input file are only '.unw','.cor','.int','.dem','.hgt_sim','.UTM_TO_RDC'.Which one is the right?I guess it's '.cor',but there is a bug : list index out of range.

> template file :
mintpy.load.connCompFile = ./../mintpy/maskConnComp.cor

> bug:
Traceback (most recent call last):
File "/home/gzhusar /tools /MintPy/mintpy/smallbaselineApp.py", line 1257, in <module>
main( sys . argv[1:])
File "/home/gzhusar /tools /MintPy/mintpy/ smallbaselineApp.py", line 1239, in main
app. run steps-inps . runSteps )
File " /home/gzhusar /tools /MintPy/mintpy /smallbaselineApp.py", line 1031, in run
self. run load data( sname )
File " /hone/gzhusar /tools /MintPy /mintpy/ smallbaselineApp.py", line 345, in run_ load data
mintpy. load data main(iargs )
File " /home /gzhusar/tools/MintPy/mintpy/load data.py", line 763, in main
prepare netadata(iDict)
File " /home/gzhusar /tools/MintPy/mintpy/load data.py", line 616, in prepare_metadata
prep_module .main(iargs)
File " /home/gzhusar/tools/MintPy /mintpy/prep_ gamma.py", line 443, in main
extract_ metadata4interferogram( fname, sensor_ name=inps . sensor . lower( ) )
File " /home/gzhusar/tools/MintPy/mintpy/prep_ gamma. py", line 240, in extract_ metadata4interferogramdate12 = str(re. findall('ld[6][-_ ]ld[6]', file basename)[0])

**Comment by yuankailiu** [comment]
https://github.com/insarlab/MintPy/blob/main/mintpy/defaults/smallbaselineApp.cfg#L35-L41

The step "load_data" is to provide the paths of your interferograms. In your case, I assume it is from Gamma?
So the connected components to be loaded are the files from Gamma, it can be either geocoded or non-geocoded files (I am not sure which file is the connected component for Gamma). 

The `maskConnComp.h5` file under the `/mintpy/` folder is the mask file based on common connected components, this is not the one we should load.

**Comment by yunjunz** [comment, marked as answer]
It seems like you are using Gamma, whose default unwrapper does not provide connected components. SNAPHU is the only unwrapper that provides them, as far as I know currrently.

**Reply by yuankailiu** [reply]
So that is why we prefer to use SNAPHU (rather than icu) in ISCE topsApp or topsStack as unwrapper? So that we can have connected components to generate a mask or identify reliable regions for Bridging unwrapping error correction.

**Reply by yunjunz** [reply]
I think that's one of the reasons. Second, SNAPHU is developed later, more advanced, and suppose to give more robust results; but slower than the simple icu algorithm.

**Comment by qingyaol** [comment]
I get it, thanks!




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zhang ***@***.***&gt;; 
发送时间: 2021年7月27日(星期二) 凌晨3:19
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [insarlab/MintPy] How to load connCompFile from Gamma products? (#591)





 
It seems like you are using Gamma, whose default unwrapper does not provide connected components. SNAPHU is the only unwrapper that provides them, as far as I know currrently.
